### ISSUE TITLE
Executed code marked as not covered

### ISSUE THREAD

**Issue by lolmaus** [sglanzer/ember-cli-blanket#105]
Here's a file:

```js
import Ember from 'ember';
import _ from 'npm:lodash';

const {
  computed
} = Ember;

export default function flatten(propName) {
  return computed(propName + '.[]', function() {
    let arrays = this.get(propName).toArray();
    return _(arrays)
      .map(a => a.toArray())
      .flatten()
      .value();
  });
}
```

In my test I do:

```js
import flatten from 'flatten';
/*...*/flatten('foo')/*...*/
```

I expect to have 100% coverage.

But here's how Blanket 0.7.0 reports the coverage:

![image](https://cloud.githubusercontent.com/assets/200884/11934491/badecc42-a813-11e5-89ab-f1f73c14d9bf.png)

This is clearly a bug. Lines 12 and 13 can not be evaluated without lines 8-10 being evaluated.

Looks like a bracket balancing bug.

**Comment by lolmaus** [comment]
Here's another example:

```js
import _ from 'npm:lodash';

export const joinUrlObj = {

  slashCharacter: '/',

  main (...urlParts) {
    return urlParts
      .filter((part) => _.isString(part) && part.length)
      .map((part, index) => this.stripSlashesFromPart(part, index, urlParts.length))
      .join(this.slashCharacter);
  },


  beginsWithSlash (str) {
    return _.startsWith(str, this.slashCharacter);
  },

  endsWithSlash (str) {
    return _.endsWith(str, this.slashCharacter);
  },

  stripSlashesFromPart (part, index, length) {
    if (index !== length - 1) {
      part = this.stripTrailingSlash(part);
    }

    if (index !== 0) {
      part = this.stripLeadingSlash(part);
    }

    return part;
  },

  stripLeadingSlash (str) {
    if (this.beginsWithSlash(str)) {
      return str.substr(1);
    }

    return str;
  },

  stripTrailingSlash (str) {
    if (this.endsWithSlash(str)) {
      return str.substr(0, str.length - 1);
    }

    return str;
  }
};

export default joinUrlObj.main.bind(joinUrlObj);
```

![image](https://cloud.githubusercontent.com/assets/200884/11935434/df6620f6-a818-11e5-87e4-75e384e49783.png)

**Comment by lolmaus** [comment]
Closed in favor of https://github.com/alex-seville/blanket/issues/546 .

**Comment by lolmaus** [comment]
So this issue is a duplicate of #100, which in turn is a regression in Ember CLI dependencies.

To work around the issue, do this: https://github.com/bmeurant/ember-array-contains-helper/commit/18ef68b67ab9172da1d74bda1029ff3f046281d4

**Comment by lolmaus** [comment]
So this issue is a duplicate of https://github.com/sglanzer/ember-cli-blanket/issues/100, which in turn is a regression in Ember CLI dependencies.

To work around the issue, do this: https://github.com/bmeurant/ember-array-contains-helper/commit/18ef68b67ab9172da1d74bda1029ff3f046281d4

**Comment by rwwagner90** [comment]
@lolmaus is this still an issue? I'm not sure if I should downgrade loader.js, especially since it has been moved to package.json now and out of bower.

**Comment by jschilli** [comment]
We released 0.9.0 yesterday which gets current with ember cli.

Unfortunately I borked the dependencies #117.  That change has been committed but @sglanzer hasn't released 0.9.1 to npm yet.

you could run against master (and I would appreciate it for verification that all of this works on apps other than mine)

**Comment by rwwagner90** [comment]
@jschilli I just updated my package.json to point to ` "ember-cli-blanket": "sglanzer/ember-cli-blanket#master"` and reinstalled, but now when I check the coverage button at the top of the tests, nothing about coverage shows up at the bottom.
![screen shot 2016-03-03 at 5 53 04 pm](https://cloud.githubusercontent.com/assets/2640861/13512459/cfb8be3c-e168-11e5-97c2-1f5705700f2d.png)

**Comment by jschilli** [comment]
@rwwagner90 what version of ember-cli? ember? 

any messages on the console?  running ember test --server?

**Comment by rwwagner90** [comment]
@jschilli I was on ember-cli 2.3.0 and ember 2.3.1 I think. I am updating to 2.4.1 now to see if that helps.

**Comment by rwwagner90** [comment]
@jschilli I see this in the console `TypeError: blanket.instrumentSync is not a function`

**Comment by jschilli** [comment]
ember g ember-cli-blanket to update the blanket version

**Comment by jschilli** [comment]
I can look @ putting some stronger feedback in around that - we could add it to the ember-cli-version-checker logic but we're running off a fork of blanket.js ATM

**Comment by rwwagner90** [comment]
@jschilli we need a specific commit of blanket?

**Comment by jschilli** [comment]
yep, 

`bower install blanket#5e94fc30f2e694bb5c3718ddcbf60d467f4b4d26`

the blueprint has been updated but doesn't run automatically unless you ember install

**Comment by jschilli** [comment]
some README updates are probably in order

**Comment by rwwagner90** [comment]
@jschilli okay, I'm at 100% coverage now. Thanks for your help :smile: . Anything I can do to help document this or get things updated so this is more automatic?

**Comment by jschilli** [comment]
awesome.

Yeah - a PR with some guidance, things to check etc in the readme would be awesome.  

I can noodle on ember-cli-version-checker in index.js - not sure if that will resolve a sha for a bower dependency (that's my recollection when I coded that)

**Comment by rwwagner90** [comment]
I'd be glad to make some README updates later, if you'd like, but it would be great if we could just get 0.9.1 out and not have to point people to master. When do you think you could publish that to npm?

**Comment by jschilli** [comment]
@sglanzer holds the keys to publish - will ping him

**Comment by jschilli** [comment]
I think even if we publish 0.9.1 those existing users of ember-cli-blanket will need to do a ember g or bower i to get it working

**Comment by rwwagner90** [comment]
@jschilli yeah definitely. I submitted a PR https://github.com/sglanzer/ember-cli-blanket/pull/120

**Comment by sglanzer** [comment]
I thought I had published 0.9.1 yesterday - it shows on npm - is the content incorrect?

**Comment by jschilli** [comment]
It is there. My bad. 

-jeff
